src/background/services/keyAutoAdd.ts

```diff
@@ -82,13 +85,16 @@ export class KeyAutoAddService {
     return this.tab?.id;
   }
 
-  private async process(url: string, payload: BeginPayload, tabId?: TabId) {
+  private async process(
+    url: string,
+    payload: BeginPayload,
+    existingTabId?: TabId,
+  ) {
     const { resolve, reject, promise } = withResolvers();
+    const tab = existingTabId
+      ? await this.browser.tabs.update(existingTabId, { url })
```

Comment:
Only concern with this approach is, what happens if tab with this ID is closed? TabId will be defined, but update call will throw.

Reply:
my bad, now I understand why `tabs.get` was used there.
let's use a `try catch` to handle that sync error that tabs.get method can throw